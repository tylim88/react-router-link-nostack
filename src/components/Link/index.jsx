import React from 'react'
import { Link, useLocation } from 'react-router-dom'

const LinkNoStack = props => {
	const { to, onClick, onSamePage, ...otherProps } = props
	const location = useLocation()

	let isSamePath = false
	try {
		isSamePath =
			location.pathname.toLowerCase() === to.toLowerCase() || to === '#'
	} catch (e) {
		console.error('(react-router-link-nostack) "to" props accept only strings')
	}
	const onClick_ = e => {
		if (isSamePath) {
			onSamePage && onSamePage()
		}
		onClick && onClick(e)
	}

	return (
		<Link to={to} onClick={onClick_} replace={isSamePath} {...otherProps} />
	)
}

export default LinkNoStack
