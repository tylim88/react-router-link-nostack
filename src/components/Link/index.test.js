import React from 'react'
import Link from './index'
import { MemoryRouter } from 'react-router-dom'
import { create } from 'react-test-renderer'

it('snapshot test', () => {
	const tree = create(
		<MemoryRouter>
			<Link to='/' />
		</MemoryRouter>
	).toJSON()
	expect(tree).toMatchSnapshot()
})
