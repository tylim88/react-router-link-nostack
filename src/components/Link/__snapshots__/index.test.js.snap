// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<a
  href="/"
  onClick={[Function]}
/>
`;
